include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -84,6 +84,7 @@ void buildAssignmentsForRegisterWrite(OpBuilder &builder,
 // A structure representing a set of ports which act as a memory interface for
 // external memories.
 struct MemoryPortsImpl {
+  StringRef memName;
```

Comment:
Can you guarantee that MemoryPortsImpl will never outlive memName? If not, I think the safe choice is just using a `std::string`.